src/IO/ForkWriteBuffer.cpp

```diff
@@ -0,0 +1,83 @@
+#include <IO/ForkWriteBuffer.h>
+#include <Common/Exception.h>
+#include <ranges>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int CANNOT_CREATE_IO_BUFFER;
+}
+
+ForkWriteBuffer::ForkWriteBuffer(WriteBufferPtrs && sources_)
+    : WriteBuffer(nullptr, 0), sources(std::move(sources_))
+{
+    if (sources.empty())
+    {
+        first_buffer = nullptr;
+        throw Exception("ForkWriteBuffer required WriteBuffer is not provided", ErrorCodes::CANNOT_CREATE_IO_BUFFER);
+    }
+    else
+    {
+        first_buffer = sources.begin()->get();
+        set(first_buffer->buffer().begin(), first_buffer->buffer().size());
+    }
+}
+
+
+void ForkWriteBuffer::nextImpl()
+{
+    if (!first_buffer)
+        return;
+
+    first_buffer->position() = position();
+
+    try
+    {
+        for (const WriteBufferPtr & write_buffer : sources | std::views::reverse)
```

Comment:
why need `std::views::reverse` here?

Reply:
Ok 👍🏻.
Just my opinion, but probably it could be better to make the this code like this:
```
        auto & source_buffer = sources.front();
        for (auto it = sources.begin() + 1; it != sources.end; ++it)
        {
            auto & buffer = *it;
            buffer->write(source_buffer->buffer().begin(), source_buffer->offset());
            buffer->next();
        }
        source_buffer.next();
```

Reply:
thanks for the suggestion. i have added the same